### DISCUSSION TITLE
Multiple environments segregated by namespaces in one cluster

### DISCUSSION THREAD

**Discussion by comminutus** [fluxcd/flux2#3471] [Q&A]
Hello, I studied the example monorepo here https://github.com/fluxcd/flux2-kustomize-helm-example and it looks like each environment specializes the kustomize base by changing the host name of the pod depending on the environment.

However, each pod in various environments is deployed to the same namespace.  I have some security concerns regarding this.  Ideally instead of only changing a pod's hostname, I'd like to change the hostname and its namespace (ex: use `podinfo.staging` for the hostname **and** the namespace at https://github.com/fluxcd/flux2-kustomize-helm-example/blob/main/apps/staging/podinfo-values.yaml).

I feel like there's some way to do this but my kustomize-fu isn't all that great.  Does anyone know how to accomplish this?  In the end, what I'm trying to accomplish is isolate several deployments by environment (dev, staging, prod), and promote these deployments as I progress in strapping them together.

Any input would be greatly appreciated!

**Comment by SrDayne** [comment]
Hello.
Just add `namespace` field to your `kustomization.yaml`. Also, kustomize do not create namespace automatically, so you should include `namespace.yaml` in resources:

**kustomization.yaml**
```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
namespace: project-dev
resources:
  - namespace.yaml
  - your_resources.yaml
```
**namespace.yaml**
```
apiVersion: v1
kind: Namespace
metadata:
  name: project-dev
```
